CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/ui/base/NavigationError.kt

```diff
@@ -0,0 +1,11 @@
+package co.nimblehq.coroutine.ui.base
+
+sealed class NavigationError(
```

Comment:
Optional: Should we keep it as `NavigationException`? Since `Exception` is more commonly used in programming, and refers to the same as `Error`. 

cc @hoangnguyen92dn @luongvo

Reply:
from this [insight](https://www.tutorialspoint.com/difference-between-exception-and-error-in-java#:~:text=Exceptions%20and%20errors%20both%20are,catch%20these%20types%20of%20problems.&text=Exceptions%20are%20the%20problems%20which%20can%20occur%20at%20runtime%20and%20compile%20time.), it should be Exception instead 💭

Reply:
Resolved accordingly